File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String longestPalindrome(String s) {
        String ans = "";
        int n = s.length();
        int resLen = 0;
        for(int i = 0;i<n;i++){
            int low = i;
            int high = i;
            while(low >= 0 && high < n && s.charAt(low) == s.charAt(high)){
                if(resLen < (high - low +1)){
                    ans = s.substring(low,high+1);//one is not takine
                    resLen = Math.max(resLen,(high-low +1));
                }
                low--;
                high++;
            }
            low = i;
            high = i+1;
            while(low >= 0 && high < n && s.charAt(low) == s.charAt(high)){
                if(resLen < (high - low +1)){
                    ans = s.substring(low,high+1);//one is not takine
                    resLen = Math.max(resLen,(high-low +1));
                }
                low--;
                high++;
            }
        }
        return ans;
    }
}

File: 5-longest-palindromic-substring/longest-palindromic-substring.c
void findSubstring(char* s,char res[],int low,int high){
    int j = 0;
    for(int i = low;i<=high;i++){
        res[j++] = s[i];
    }
    res[j] ='\0';
}
char* longestPalindrome(char* s) {
    int n = strlen(s);
    static char res[1001] = {0};
    int resLen = 0;
    for(int i= 0;i<n;i++){
        int low = i;
        int high = i;
        while(low >= 0 && high < n&& s[low] ==  s[high]){
            int len = high - low +1;
            if(resLen < (len)){
                findSubstring(s,res,low,high);
                resLen = len ;
            }
            low--;
            high++;
        }
        low = i;
        high = i+1;
        while(low >= 0 && high < n&& s[low] ==  s[high]){
            int len = high - low +1;
            if(resLen < (len)){
                findSubstring(s,res,low,high);
                resLen = len;
            }
            low--;
            high++;
        }
    }
    return res;
}